include/deal.II/lac/la_parallel_vector.h

```diff
@@ -1243,7 +1298,10 @@ namespace LinearAlgebra
        * we need to use a custom deleter for this object that does not call
        * <code>delete[]</code>, but instead calls @p free().
        */
-      std::unique_ptr<Number[], decltype(&free)> values;
+      // TODO remove mutable when update_ghost is changed
+      mutable std::unique_ptr<Number[], decltype(&free)> values;
+
+      Number *values_dev;
```

Comment:
is `dev` "device" or "developer"?

And is it correct that you don't use a `unique_ptr` because that class is not available on the device? (If so, what a pain!)

Reply:
> And is it correct that you don't use a unique_ptr because that class is not available on the device? (If so, what a pain!)

We can use a `std::unique_ptr` in combination with `CUDA` memory.
We just have to allocate memory using `cudaMalloc` and pass the pointer to a `std::unique_ptr` with a suitable deleter that calls `cudaFree` (see https://stackoverflow.com/questions/47383386/use-of-unique-ptr-and-cudamalloc/47406068#47406068).
Of course, the type for `Host` and `CUDA` memory will still be different.